src/App.tsx

```diff
@@ -14,11 +17,85 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [goodsList, setGoodsList] = useState(goodsFromServer);
+  const [isVisibleStart, setVisibleStart] = useState(true);
+
+  const hideStart = () => {
+    setVisibleStart(prev => !prev);
+  };
```

Comment:
you can read about [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback)